### PR TITLE
mobile: add MapWidget.qrc and solve #1410

### LINKED ISSUE #1410: mobile: Dive flags are missing from the maps
### Describe the issue:

- [x] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:

Dive flags are missing from the maps.

As reported by John Smith. I can reproduce.

### Operating system:

iOS: 11.3.1
App Version: 2.1.0 (4.7.8.372) TestFlight

### PR DESCRIPTION
**Pull request by janiversen**
move aliases to mobile-resources.qrc

### Describe the pull request:

- [ x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

add MapWidget.qrc to clean mobile-resources.qrc and subsurface.qrc
and adapt builds.

iOS uses only mobile-resources.arc and MapWidget.qrc, whereas
Android uses mobile-resources.qrc, subsurface.qrc and MapWidget.qrc

Desktop uses subsurface.qrc and MapWidget.qrc
mobile uses mobile-resources.qrc, subsurface.qrc and MapWidget.qrc

fixes #1410
### Mentions:

@mfbernardes can you please test ?

### CODE REVIEW

File: mobile-widgets/qml/mobile-resources.qrc
Comment (neolit123) [inline mobile-widgets/qml/mobile-resources.qrc]: indentation issues. but this also doubles some aliases for the same file names.
see `mapwidget-marker*` above.

i think this commit 8ef87e618a was a bad idea, @dirkhh .
@janiversen the mobile and desktop version need to use the same QRC aliases as those in the QML for the map widget - MapWidget.qml.

given this build from a couple of days ago just worked fine on my Android device:
https://github.com/Subsurface-divelog/subsurface/releases/tag/continuous

my only guess is that we are using desktop resources for the Android mobile build. o_O.

Comment (janiversen) [reply mobile-widgets/qml/mobile-resources.qrc]: you are correct, put this PR on hold, but at least I know what the problem. Will continue work tomorrow.

Comment (janiversen) [reply mobile-widgets/qml/mobile-resources.qrc]: @neolit123 just checked mapwidget/qml should have a qrc file instead of having defintions in both subsurface.qrc and mobile-widgets.qrc. I need some hours sleep but then I will update this PR.

Comment (neolit123) [reply mobile-widgets/qml/mobile-resources.qrc]: sounds good.

Comment (dirkhh) [reply mobile-widgets/qml/mobile-resources.qrc]: I like the idea of one qrc file that's shared between mobile and desktop for the map
Also, @janiversen when you do a commit that fixes an issue, if you add a line like this in the commit message itself (not the PR text, the commit message)
```
Fixes #1410
```
Then GitHub closes the issue automagically.
Finally... please have a Subsurface config for your editor / IDE so that by default you don't mangle whitespace? It must be frustrating for you to always be reminded by the rest of us...

Comment (janiversen) [reply mobile-widgets/qml/mobile-resources.qrc]: thanks, I have my editors configured, but copy/paste on a mac does strange things.

**Comment by neolit123** [comment]
FYI i have a suspicion that our Android build is using the desktop QRC for the map assets. while the iOS one is using the mobile QRC for the map assets. not sure what's going on, but this was the result of yesterday's tests.

**Comment by janiversen** [comment]
that is a correct observation, I decoupled iOS from the desktop qrc a while ago. I am right now trying to figure out how the android build is configured (to add the new map-widget.qrc).

**Comment by mfbernardes** [comment]
On a quick look it seems the CMakeLists.txt
 just adds subsurface.qrc

If I’m not mistaken cmake is also used for Android, so things get mixed.

But since I never check the android build in details I might be wrong

**Comment by janiversen** [comment]
hmmm I did add MapWidget.qrc and thought it was unconditionally. Let me have a second look

**Comment by janiversen** [comment]
I add it in line 272 which seems to be unconditionally, at least I cannot find lines the removes already defined resources. Line 306 and 207 adds (extra) resources for android and the mobile versions. I have tested desktop, mobile and iOS, but I cannot test on Android today.

File: map-widget/qml/MapWidget.qrc
Comment (dirkhh) [inline map-widget/qml/MapWidget.qrc]: mostly just curious - why is there this section of resources that are commented out?

**Review by dirkhh — APPROVED** [review]
I think this looks good and it fixes a bug. Yay.
Happy to merge it when I'm actually awake (it's 4am - darn insomnia). Just a small question on the first commit...

File: map-widget/qml/MapWidget.qrc
Comment (janiversen) [reply map-widget/qml/MapWidget.qrc]: That was while I tested, let me remove that fast.

File: packaging/ios/Subsurface-mobile.pro
Comment (neolit123) [inline packaging/ios/Subsurface-mobile.pro]: the filename is `MapWidget.qrc` but here `mapWidget.qrc` is used, would that be a problem?

also, why not name the file lowercase as per the Qt standard? `map-widget.qrc`

Comment (dirkhh) [reply packaging/ios/Subsurface-mobile.pro]: yeah, I'd prefer all lowercase...
IIRC Mac still allows for a filesystem to not be case sensitive.

**Comment by dirkhh** [comment]
tested Android build and that works fine as well. Great work @janiversen

**Pull request merged**
